python/orca/test/bigdl/orca/learn/ray/tf/test_tf_spark_estimator.py

```diff
@@ -126,7 +125,7 @@ def test_dataframe_with_empty_partition(self):
                 model_creator=model_creator,
                 verbose=True,
                 config=config,
-                workers_per_node=2,
+                workers_per_node=3,
```

Comment:
After changing num_workers to 3, the unit test will fail. 4 partitions with an empty partition repartition to 3 will still have an empty partition.
```
>>> rdd = sc.parallelize([1, 2, 3], numSlices=4)
>>> rdd.glom().collect()
[[], [1], [2], [3]]                                                        	 
>>> rdd2 = rdd.repartition(3)
>>> rdd2.glom().collect()
[[], [2], [1, 3]]
```
Please take a look @jenniew

Reply:
We'd better do repartition before converting to SparkXShards to avoid such issue.